controllers/remote/cluster_cache.go

```diff
@@ -0,0 +1,203 @@
+/*
```

Comment:
Maybe we rename this to cluster_cache_controller?

Reply:
See my comment below about manager vs reconciler. I'd prefer to keep this as-is, or rename to cluster_cache_manager.go. But again, if you feel strongly, I can rename it.